src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6322,12 +6561,190 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
 
         public override IOperation? VisitInterpolatedStringHandlerCreation(IInterpolatedStringHandlerCreationOperation operation, int? captureIdForResult)
         {
-            return VisitNoneOperation(operation, captureIdForResult);
+            // We turn the interpolated string into a call to create the handler type, a series of append calls (potentially with branches, depending on the
+            // handler semantics), and then evaluate to the handler flow capture temp.
+
+            SpillEvalStack();
+            RegionBuilder resultRegion = CurrentRegionRequired;
+            var handlerCaptureId = captureIdForResult ?? GetNextCaptureId(resultRegion);
+            var previousHandlerId = _currentInterpolatedStringHandlerId;
+            _currentInterpolatedStringHandlerId = handlerCaptureId;
```

Comment:
>_currentInterpolatedStringHandlerId [](http://example.com/codeflow?start=12&length=35)

This can be moved to the proposed interpolation context as a mutable state.